sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1458,6 +1458,15 @@ object SQLConf {
       .booleanConf
       .createWithDefault(true)
 
+  val JSON_EXPRESSION_OPTIMIZATION =
+    buildConf("spark.sql.optimizer.enableJsonExpressionOptimization")
```

Comment:
hmm, I think users can already disable any optimizer rule via `spark.sql.optimizer.excludedRules`?

Reply:
`excludedRules` should work, but I'm thinking to add CSV expression optimization similar to JSON in same rule. If so, a separate config for JSON/CSV expression seems good. If we have two rules for JSON, CSV individually, then `excludedRules` might be enough.

Reply:
I see, makes sense.